src/upsample.jl

```diff
@@ -234,25 +258,72 @@ function ChainRulesCore.rrule(::typeof(upsample_bilinear), x; size)
 end
 
 """
-    pixel_shuffle(x, r)
+    pixel_shuffle(x, r::Integer)
+
+Pixel shuffling operation, upscaling by a factor `r`.
+The operation converts an input of size [W,H,r²C,N] to size [rW,rH,C,N].
```

Comment:
Can we state somehow that pixel_shuffle supports arbitrary dimensions?

Reply:
I put an example of a "1D image" , but perhaps these words should change too.